packages/contracts/contracts/voting/GovernancePlugin.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import {Counters} from "@openzeppelin/contracts/utils/Counters.sol";
+
+import {Plugin} from "../core/plugin/Plugin.sol";
+import {IDAO} from "../core/IDAO.sol";
+import {IProposal} from "./IProposal.sol";
+import {GovernanceBase} from "./GovernanceBase.sol";
+
+abstract contract GovernancePlugin is GovernanceBase, Plugin {
+    using Counters for Counters.Counter;
+
+    /// @notice The incremental ID for proposals and executions.
+    Counters.Counter private proposalCounter;
+
+    /// @notice The [ERC-165](https://eips.ethereum.org/EIPS/eip-165) interface ID of the contract.
+    bytes4 internal constant GOVERNANCE_INTERFACE_ID = type(IProposal).interfaceId;
+
+    /// @notice Checks if this or the parent contract supports an interface by its ID.
+    /// @param interfaceId The ID of the interace.
+    /// @return bool Returns true if the interface is supported.
+    function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
+        return interfaceId == GOVERNANCE_INTERFACE_ID || super.supportsInterface(interfaceId);
+    }
+
+    /// @inheritdoc IProposal
+    function proposalCount() external view returns (uint256) {
```

Comment:
The idea was to use OZ upgradeable and non-upgradeable contracts for the respective plugin type.
Since the counter consists of a library, the upgradeable and non-upgradeable are exactly the same and the suggestion by the reviewer could be done.

However, after a discussion with @novaknole and since the `proposalId` will most likely change anyway, we will tackle these organizational changes later and in line with the outcome of https://github.com/aragon/core/pull/155.

Reply:
1. Done in 6f81b9edcdcebe1791d918938d5c3c4fb1d16017
2. Done in 635b44d2ef9b9cb6c558e4f20358f6befb7d3add